dev_guide/builds.adoc

```diff
@@ -846,43 +850,48 @@ source:
 
 == Binary Source
 
-When the `*BuildConfig.spec.source.type*` is `*Binary*`, the build will expect a
-binary as input, and an inline Dockerfile is optional.
-
-The binary is generally assumed to be a tar, gzipped tar, or zip file depending
-on the strategy. For `*Docker*` builds, this is the build context and an
-optional Dockerfile may be specified to override any Dockerfile in the build
-context. For `*Source*` builds, this is assumed to be an archive as described
-above. For `*Source*` and `*Docker*` builds, if `*binary.asFile*` is set the
-build context will consist of a single file named by the value of `*binary.asFile*`. 
-The `*contextDir*` field may be used when an archive is provided. Custom builds will 
-receive this binary as input on standard input (`stdin`).
+Streaming content in binary format from a local file system to the builder is called a `*binary type build*`.
+The corresponding value of `*BuildConfig.spec.source.type*` is `*Binary*` for such builds.
 
-A binary source potentially extracts content (if `*asFile*` is not set), in which case `*contextDir*`
-allows changing to a subdirectory within the content before the build executes.
-
-The source definition is part of the `*spec*` section in the `*BuildConfig*`:
+This source type is unique in that it is leveraged solely based on your use of the `oc start-build`.
 
+[NOTE]
 ====
-[source,yaml]
-----
-source:
-  type: "Binary"
-  binary: <1>
-    asFile: "webapp.war" <2>
-  contextDir: "app/dir" <3>
-  dockerfile: "FROM centos:7\nRUN yum install -y httpd" <4>
-----
-<1> The `*binary*` field specifies the details of the binary source.
-<2> The `*asFile*` field specifies the name of the file that will be created with
-the binary contents.
-<3> The `*contextDir*` field specifies a subdirectory within the extracted contents of a
-binary archive.
-<4> If the optional `*dockerfile*` field is provided, it should be a string
-containing an inline Dockerfile that potentially replaces one within the
-contents of the binary archive.
```

Comment:
> The problem is we reference *binary.asFile*, *dockerfile* and *contextDir* down below and these things relate to nothing concrete in the context; they're like null pointers, aren't they?

the binary.asFile reference could/should be removed.  the dockerfile and contextDir references are referencing to concepts/fields described elsewhere in the doc, they are not specific not binary build, so no, they aren't null pointers.

The only argument in my head for showing how to define a BuildConfig with a Binary build source is because it's possible a user might want to intentionally define a buildconfig that they only intend to build via the binary process (therefore they do not have a valid Dockerfile, Git, or Image source they can define as part of the BuildConfig).

However even if we instruct them on how to do that, we should not be explaining asFile because that is still a field they have zero control over (it doesn't matter if they provide it or not, or what value they provide, it always gets overwritten).

so in sum i'm ok w/ putting the example back, but with modifications.